room_with_a_view/room_with_a_view.py

```diff
@@ -1,3 +1,397 @@
+        views that can be reached from the views we're syncing, then recreate
+        the views in topological order on the subgraph using Kahn's algorithm
+        (as we do in ``sync_all()``).
+        """
+        statement_names = self.options.view_names
+        file_names = self.options.file_names
+        if not statement_names and not file_names:
+            raise ValueError('Either --view-names or --file-names is required '
+                             'for the "drop" action.')
+
+        for file_name in file_names:
+            statement_names.extend(self.get_statements_from_file(file_name))
+
+        for statement_name in statement_names:
+            if statement_name not in self.dependency_graph:
+                raise ValueError('unrecognized view or function name: {}'
+                                 .format(statement_name))
+
+        # First, build a graph containing only nodes reachable from the views
+        # and functions to sync.
+        starting_nodes = [self.dependency_graph[statement_name]
+                          for statement_name in statement_names]
+        subgraph_node_names = self.traverse_graph(starting_nodes,
+                                                  dependency_order=False)
+        subgraph = {}
+        for node_name in subgraph_node_names:
+            original_node = self.dependency_graph[node_name]
+            new_node = DependencyGraphNode(
+                name=original_node.name,
+                declaration=original_node.declaration,
+                statement_type=original_node.statement_type,
+                body=original_node.body,
+                comments=original_node.comments,
+                arg_list=original_node.arg_list)
+            subgraph[node_name] = new_node
+            new_node.in_edges = set([edge for edge in original_node.in_edges
+                                     if edge in subgraph_node_names])
+            new_node.out_edges = set([edge for edge in original_node.out_edges
+                                      if edge in subgraph_node_names])
+
+        # Then, drop all the views and functions.
+        for node in starting_nodes:
+            self.drop_node(node)
+
+        # Finally, run Kahn's algorithm to recreate the subgraph in dependency
+        # order.
+        starting_nodes = [node for node in subgraph.values()
+                          if not node.out_edges]
+        self.traverse_graph(starting_nodes, graph=subgraph,
+                            visit_function=self.create_node)
+
+    def sync_all(self):
+        """ Syncs all views and functions to Redshift. """
+        starting_nodes = [node for node in self.dependency_graph.values()
+                          if not node.out_edges]
+        self.traverse_graph(
+            starting_nodes, visit_function=self.drop_and_recreate_node)
+
+    def traverse_graph(self, starting_nodes, graph=None, visit_function=None,
```

Comment:
Why does every open source project i work on have a topsort implementation? https://github.com/b12io/orchestra/blob/dbd32fd243269150195166b79365810f246a46a1/orchestra/project_api/api.py#L69

Reply:
Because the world is a graph, and networkx has too many algorithms in it that you never need.